parser/lexer.go

```diff
@@ -285,70 +285,107 @@ func scanQuotedIdent(s *Scanner) (tok int, pos Pos, lit string) {
 }
 
 func startString(s *Scanner) (tok int, pos Pos, lit string) {
-	return s.scanString()
+	tok, pos, lit = s.scanString()
+
+	// Quoted strings placed next to each other are concatenated to a single string.
+	// See http://dev.mysql.com/doc/refman/5.7/en/string-literals.html
+	ch := s.skipWhitespace()
+	for ch == '\'' || ch == '"' {
+		_, _, lit1 := s.scanString()
+		lit = lit + lit1
+		ch = s.skipWhitespace()
+	}
+	return
+}
+
+// lazyBuf is used to avoid buf if possible.
+type lazyBuf struct {
+	flag bool
+	r    *reader
+	b    *bytes.Buffer
+	p    *Pos
+}
+
+func (mb *lazyBuf) useBuf(str string) {
+	if !mb.flag {
+		mb.flag = true
+		mb.b.Reset()
+		mb.b.WriteString(str)
+	}
+}
+
+func (mb *lazyBuf) writeRune(r rune) {
+	if mb.flag {
+		mb.b.WriteRune(r)
+	}
+}
+
+func (mb *lazyBuf) data() string {
+	var lit string
+	if mb.flag {
+		lit = mb.b.String()
+	} else {
+		lit = mb.r.data(mb.p)
+		lit = lit[1 : len(lit)-1]
+	}
+	return lit
```

Comment:
it's here...  
if mb.flag is false.... scanString is a zero alloc function
@coocood